azurecontainerapps.ts

```diff
@@ -233,12 +230,46 @@ export class azurecontainerapps {
         // Set deployment location, if provided
         let location: string = this.toolHelper.getInput('location', false);
 
-        // If no location was provided, use the default location for the Container App service
-        if (this.util.isNullOrEmpty(location)) {
-            location = await this.appHelper.getDefaultContainerAppLocation();
+        if (!this.util.isNullOrEmpty(location)) {
+            return location;
+        }
+
+        // If no location was provided, attempt to discover the location of the existing Container App Environment linked to the Container App
+        // or Container App Environment provided in the resource group or use the default location.
+        let resourceGroup: string = this.toolHelper.getInput('resourceGroup', false);
+
+        // Get the resource group if it was provided, or generate it from the Container App name
+        !this.util.isNullOrEmpty(resourceGroup) ? resourceGroup : `${this.containerAppName}-rg`;
```

Comment:
I'm a bit confused about generating a new resource group name. In the context of this method, we're looking for the location. In this context, if the resource group isn't available, then we know already that we should just return the default location, right? I'm not clear in which case does having a generated resource group name help us.